modules/linear_boltzmann_solvers/executors/pi_keigen_smm.cc

```diff
@@ -0,0 +1,1122 @@
+        {
+          const auto g = first_grp + gsg;
+
+          // Get the tensor, set it to zero
+          const auto dof_map = pwld.MapDOFLocal(cell, i, tensor_uk_man_, g, 0);
+          double* T = &local_tensors[dof_map];
+
+          // Perform the quadrature integration
+          for (int n = 0; n < num_gs_angles; ++n)
+          {
+            const auto& omega = quadrature->omegas_[n];
+            const auto& wt = quadrature->weights_[n];
+
+            const auto psi_dof = pwld.MapDOFLocal(cell, i, psi_uk_man, n, gsg);
+            const auto coeff = wt * psi[gs][psi_dof];
+
+            for (int k = 0; k < dimension_; ++k)
+            {
+              for (int l = 0; l < dimension_; ++l)
+                T[k * dimension_ + l] +=
+                  coeff * omega[dimension_ > 1 ? k : 2] * omega[dimension_ > 1 ? l : 2];
+              T[k * dimension_ + k] -= coeff / 3.0;
+            }
+          } // for angle n
+        }   // for groupset group gsg
+      }     // for node i
+
+      // Loop over cell faces
+      int f = 0;
+      for (const auto& face : cell.faces_)
+      {
+        if (not face.has_neighbor_)
+        {
+          const auto num_face_nodes = cell_mapping.NumFaceNodes(f);
+          for (int fi = 0; fi < num_face_nodes; ++fi)
+          {
+            const auto i = cell_mapping.MapFaceNode(f, fi);
+            const auto imap = pwld.MapDOFLocal(cell, i);
+
+            // Compute the diffusion boundary factor. This is simply the
+            // computation of the quadrature of \Omega \cdot \hat{n} divided
+            // by the sum of the quadrature weights. In most cases, this should
+            // be equivalent to half.
+            if (bndry_factors_.count(imap) == 0)
+            {
+              bndry_factors_[imap].resize(num_groupsets, 0.0);
+
+              double val = 0.0;
+              for (int n = 0; n < num_gs_angles; ++n)
+                val +=
+                  quadrature->weights_[n] * std::fabs(quadrature->omegas_[n].Dot(face.normal_));
+              bndry_factors_[imap][gs] = val / wt_sum;
+            }
+            //            const auto& bfac = bndry_factors_[imap][gs];
+
+            // Initialize corrections, if not done already
+            betas_[imap].assign(lbs_solver_.NumGroups(), 0.0);
+            auto& beta = betas_[imap];
+
+            // Compute the boundary corrections
+            for (int gsg = 0; gsg < num_gs_groups; ++gsg)
+            {
+              const auto g = first_grp + gsg;
+
+              // Perform the quadrature integration
+              for (int n = 0; n < num_gs_angles; ++n)
+              {
+                const auto& wt = quadrature->weights_[n];
+                const auto& omega = quadrature->omegas_[n];
+                const auto mu = std::fabs(omega.Dot(face.normal_));
+
+                const auto psi_dof = pwld.MapDOFLocal(cell, i, psi_uk_man, n, gsg);
+
+                //                beta[g] += wt * (mu - bfac) * psi[gs][psi_dof];
```

Comment:
same as before I suspect. However, I thought that the ang quadrature was not exact for half-range integrals, so why are you using 0.5 here, and not the approximate values fromt he ang quadrature? Or am I missing the mark completely here?

Reply:
You are not missing the mark at all. You just caught a very small cheat that I have that I intended to fix but never got around to. Because we are recycling the PWLC diffusion code, the boundary conditions in the matrix are set with 0.5. For the sake of consistency, I use 0.5 here as well. The lines that are commented out for this reason, but should remain for whenever the diffusion boundary condition is changed in the matrix.

Reply:
@ragusa The latest commit implements the correct boundary conditions. I can discuss it more in depth tomorrow at our meeting as that it is a little convoluted. Nonetheless, I'm sure you'd be interested to know that the final eigenvalue did not change through at least the 6th digit.